File: app/src/main/java/com/example/asonbom/data/AuthInterceptor.java
package com.example.asonbom.data;

import com.google.gson.Gson;

public class AuthInterceptor {
    public  String json;
    public void params(Object data){
        Gson gson = new Gson();
        String jsonString = gson.toJson(data);
        json = jsonString;
    }
}

File: app/src/main/java/com/example/asonbom/data/AuthInterceptor.kt
package com.example.asonbom.data

import android.content.Context
import com.example.asonbom.utils.SessionManager
import okhttp3.Interceptor
import okhttp3.Response

class AuthInterceptor(context: Context) : Interceptor{

    private val sessionManager = SessionManager(context)

    override fun intercept(chain: Interceptor.Chain): Response {
        val requestBuilder = chain.request().newBuilder()

        sessionManager.fetchAuthToken()?.let{
            requestBuilder.addHeader("Autorization","Bearer $it")
        }

        return chain.proceed(requestBuilder.build())
    }
}